src/index.ts

```diff
@@ -16,7 +16,7 @@ import { fastRelativePath, isNull, isString, isUndefined, negate, pluralize, tri
 import type { FormatOptions, Options, PluginsOptions } from "./types.js";
 
 async function run(options: Options, pluginsDefaultOptions: PluginsOptions, pluginsCustomOptions: PluginsOptions): Promise<void> {
-  if (options.globs.length || !isString(await getStdin())) {
+  if (options.globs.length || (!isString(await getStdin()) && !("stdinFilepath" in options))) {
```

Comment:
can you explain this one?

if there is stdin and there is a `stdinFilepath`, we now `runGlobs`?

and if there's no stdin but there is a `stdinFilepath`, we'd also `runGlobs`?

unless im reading wrong, thats what its doing now

Reply:
I based this branch on yours, you seem to have added this condition in https://github.com/prettier/prettier-cli/pull/58/commits/585f1be4de383177350106b52125c459d180903b. Not really sure what the rationale was. 😄